### PR TITLE
Add error message to SyncState for failed sync down

### PR DESCRIPTION
**Pull request by booker0108**
Set message from error to current sync state
Return sync state to react native instead of "Sync failed" for sync failure

### CODE REVIEW

**Comment by booker0108** [comment]
Hi SFDC dev team,

Please let me know if this change is good to be merged

If yes, I can do the same for iOS SDK too

File: libs/SalesforceReact/src/com/salesforce/androidsdk/reactnative/bridge/SmartSyncReactBridge.java [hunk 1]
```diff
@@ -250,7 +250,10 @@ private void handleSyncUpdate(final SyncState sync, Callback successCallback, Ca
                     ReactBridgeHelper.invokeSuccess(successCallback, sync.asJSON());
                     break;
                 case FAILED:
-                    errorCallback.invoke("Sync failed");
+                    //errorCallback.invoke("Sync failed");
```
Comment (wmathurin) [inline libs/SalesforceReact/src/com/salesforce/androidsdk/reactnative/bridge/SmartSyncReactBridge.java]: Could you get rid of that commented code?

File: libs/SalesforceReact/src/com/salesforce/androidsdk/reactnative/bridge/SmartSyncReactBridge.java [hunk 2]
```diff
@@ -250,7 +250,10 @@ private void handleSyncUpdate(final SyncState sync, Callback successCallback, Ca
...
+
+                    //Return sync to React Native with the error message in the JSON
+                    ReactBridgeHelper.invokeSuccess(errorCallback, sync.asJSON());
```
Comment (wmathurin) [inline libs/SalesforceReact/src/com/salesforce/androidsdk/reactnative/bridge/SmartSyncReactBridge.java]: Could you rename ReactBridgeHelper.invokeSuccess methods to ReactBridgeHelper.invoke and rename also the first argument to those methods from successCallback to callback?

File: libs/SmartSync/src/com/salesforce/androidsdk/smartsync/util/SyncState.java [hunk 3]
```diff
@@ -200,6 +205,7 @@ public static SyncState fromJSON(JSONObject sync) throws JSONException {
         state.maxTimeStamp = sync.optLong(SYNC_MAX_TIME_STAMP, -1);
 		state.startTime = sync.optLong(SYNC_START_TIME, 0);
 		state.endTime = sync.optLong(SYNC_START_TIME, 0);
+		state.errorJSON = sync.getString(SYNC_ERROR);
```
Comment (wmathurin) [inline libs/SmartSync/src/com/salesforce/androidsdk/smartsync/util/SyncState.java]: Could you instead do
```java
state.errorJSON = JSONObjectHelper.optString(sync, SYNC_ERROR, "");
```
Sync states created before the code change will be missing that field, and calling getString(...) will throw a JSONException.

File: libs/SalesforceReact/src/com/salesforce/androidsdk/reactnative/bridge/SmartSyncReactBridge.java (hunk 1 above)
Comment (booker0108) [reply libs/SalesforceReact/src/com/salesforce/androidsdk/reactnative/bridge/SmartSyncReactBridge.java]: Resolved in 5f766a5e25f7e5a2c4a1956e038e4dc5c31e8881

File: libs/SalesforceReact/src/com/salesforce/androidsdk/reactnative/bridge/SmartSyncReactBridge.java (hunk 2 above)
Comment (booker0108) [reply libs/SalesforceReact/src/com/salesforce/androidsdk/reactnative/bridge/SmartSyncReactBridge.java]: Resolved in c2d0a02bb179ceda2f52773f03a13e4e19fed34f

File: libs/SmartSync/src/com/salesforce/androidsdk/smartsync/util/SyncState.java (hunk 3 above)
Comment (booker0108) [reply libs/SmartSync/src/com/salesforce/androidsdk/smartsync/util/SyncState.java]: Resolve in c1f40ddfd48b9c8cfde422d33a880c4d559244e8

**Pull request merged**